pandas/core/internals.py

```diff
@@ -5457,6 +5471,12 @@ def get_empty_dtype_and_na(join_units):
         if blk is None:
             return np.float64, np.nan
 
+    if is_uniform_reindex(join_units):
+        # XXX: integrate property
```

Comment:
@TomAugspurger long shot any recollection what this comment is asking for?

Reply:
Removing the need for that `is_uniform_reindex` check by including that case in the main `if / elif` block (which may be done on master now?)